### PR TITLE
Implement removeSubscription method

### PR DESCRIPTION
**Pull request by tiamo**

### CODE REVIEW

File: lib/src/client.dart [hunk 1]
```diff
@@ -154,9 +154,12 @@ class ClientImpl implements Client, GeneratedMessageSender {
   }
 
   @override
-  @alwaysThrows
   Future<void> removeSubscription(Subscription subscription) async {
-    throw UnimplementedError;
+    if (subscription != null) {
+      String _channel = subscription.channel;
```
Comment (mogol) [inline lib/src/client.dart]: Please update to `final channel`, see [effective-dart](https://dart.dev/guides/language/effective-dart/style#dont-use-a-leading-underscore-for-identifiers-that-arent-private)

**Comment by mogol** [comment]
@tiamo thanks for your interest and contribution!
I would like to ask you to add a few unit tests for success/failure cases and ready to merge it :+1:

File: lib/src/client.dart (hunk 1 above)
Comment (tiamo) [reply lib/src/client.dart]: ok, got it

**Comment by tiamo** [comment]
> @tiamo thanks for your interest and contribution!
> I would like to ask you to add a few unit tests for success/failure cases and ready to merge it 👍

Okay, I'll do it soon.

**Comment by tiamo** [comment]
@mogol please check

File: lib/src/client.dart [hunk 2]
```diff
@@ -58,13 +58,18 @@ abstract class Client {
   ///
   void disconnect();
 
+  /// Detect that the subscription already exists.
+  ///
+  bool hasSubscription(String channel);
```
Comment (mogol) [inline lib/src/client.dart]: 👍

**Comment by FZambia** [comment]
Thanks! Will try to make new release very soon.

**Comment by FZambia** [comment]
0.4.1 released

**Pull request merged**